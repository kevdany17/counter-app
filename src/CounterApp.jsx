import { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value)

    const handleAdd = () => setCounter( counter + 1 )

    const handleSubstract = () => setCounter( counter - 1 )

    const handleReset = () => setCounter( value )

    return (<>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button aria-label='btn-decrement' onClick={ handleSubstract }> -1 </button>
        <button aria-label='btn-increment' onClick={ handleAdd }> +1 </button>
        <button data-testid="btn-reset" onClick={ handleReset }> Reset </button>
    </>)
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaulProps = {
    value: 0
}

export default CounterApp;
