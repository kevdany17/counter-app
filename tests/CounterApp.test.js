import { render, screen, fireEvent } from "@testing-library/react";
import CounterApp from "../src/CounterApp";
import { element } from "prop-types";

describe('Counter App Test', ()=>{
    test('should show <CounterApp /> correctly', ()=>{
        const { container } = render(<CounterApp value={ 0 } />);
        expect(container).toMatchSnapshot();
    })

    test('should show the default value of 100', () => { 
        render(<CounterApp value={ 100 } />);
        const element = screen.getByRole("heading", { level: 2});
        expect(element.innerHTML).toContain("100");
    });

    test('should increment counter on click', () => { 
        render(<CounterApp value={ 0 } />);
        fireEvent.click( screen.getByRole("button", { name: "btn-increment"}))
        const element = screen.getByRole("heading", { level: 2});
        expect(element.innerHTML).toContain("1");
    });

    test('should decrement counter on click', () => { 
        render(<CounterApp value={ 0 } />);
        fireEvent.click( screen.getByRole("button", { name: "btn-decrement"}))
        const element = screen.getByRole("heading", { level: 2});
        expect(element.innerHTML).toContain("-1");
    });

    test('should reset counter to zero', ()=>{
        render(<CounterApp value={ 0 } />);
        fireEvent.click( screen.getByTestId("btn-reset") );
        const element = screen.getByRole("heading", { level: 2});
        expect(element.innerHTML).toContain("0");
    });
});