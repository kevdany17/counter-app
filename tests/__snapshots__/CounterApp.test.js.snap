// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter App Test should show <CounterApp /> correctly 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    0
     
  </h2>
  <button
    aria-label="btn-decrement"
  >
     -1 
  </button>
  <button
    aria-label="btn-increment"
  >
     +1 
  </button>
  <button
    data-testid="btn-reset"
  >
     Reset 
  </button>
</div>
`;
